### PR TITLE
How can you eat a full-course meal through a hardsuit?

### PR DESCRIPTION
**Pull request by TheDiode**
## What Does This PR Do
You can no longer eat if wearing a mask or headgear covering your mouth.

## Why It's Good For The Game
Better RP, more immersive. It is ridiculous seeing people wearing spacesuits and gas masks eating things. This also means you cannot just eat healing pills while running and wearing a gas mask, you need to stop and take it off to eat.

This does not apply to:
Vox breath mask, plasmaman helmets, clown and mime masks.

## Changelog
:cl:
tweak: You can no longer eat while wearing a mask or helmet covering your mouth. Take it off and enjoy the fresh station air.
/:cl:

### CODE REVIEW

**Comment by Bm0n** [comment]
I am pretty sure this was something that was purposely removed so that Vox and Plasmamen could eat

**Comment by Inadvizable** [comment]
I like this in theory. I realized that this probably kills vox but I like that.
Jokes aside that and plasmamen need to be handled somehow for this to be a viable idea.

**Comment by Mitchs98** [comment]
Doesn't this affect ALL forms of masks, including bandanas over the mouth?

If so, why? It's not far-fetched to push food under a bandana. And there are some items that go on the mask/face slot that shouldn't count either. Such as cigarettes.

Unless I misread the code and such masks already won't/don't count?

**Comment by Mitchs98** [comment]
> @Mitchs98 If the mask has MASKCOVERSMOUTH you can't eat through it, same with HEADCOVERSMOUTH for headgear.

THen I really disagree with it. There are several mask items, as mentioned, that make sense to eat through. I believe bandanas over mouth have that flag? Which is a very key example of it.

**Comment by Mitchs98** [comment]
> @Mitchs98 You can adjust bandanas, like breath masks, with an action button. You don't have to take those off.

Shouldn't have to. It makes 0 sense if you're trying to go for immersion here, which this PR is. You can drink and eat by pushing food under a bandana perfectly.

This also has implications for administering pills during surgery and medical stuff. 

Not to mention large balance concerns for antags trying to medicate themselves on the run, etc.

I just don't agree that a very tiny amount of immersion is worth so many anti-QOL and balance concerns.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1080,6 +1080,16 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 /mob/living/carbon/proc/eat(obj/item/reagent_containers/food/toEat, mob/user, bitesize_override)
 	if(!istype(toEat))
 		return 0
+	if(iscarbon(user))
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/carbon.dm:1083]: this check is unnecessary, this proc should never be called on anything other than a carbon mob

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -1080,6 +1080,16 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
+		var/mob/living/carbon/carbon_user = user
+		var/covered = ""
+		if(carbon_user.is_mouth_covered(head_only = TRUE))
+			covered = "headgear"
+		else if(carbon_user.is_mouth_covered(mask_only = TRUE))
+			covered = "mask"
+		if(covered)
+			to_chat(carbon_user, "<span class = 'warning'>You have to remove your [covered] first!</span>")
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/carbon.dm:1091]:
```suggestion
			to_chat(carbon_user, "<span class ='warning'>You have to remove your [covered] first!</span>")
```

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -1080,6 +1080,16 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 /mob/living/carbon/proc/eat(obj/item/reagent_containers/food/toEat, mob/user, bitesize_override)
 	if(!istype(toEat))
 		return 0
+	if(iscarbon(user))
+		var/mob/living/carbon/carbon_user = user
+		var/covered = ""
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/carbon.dm:1085]:
```suggestion
		var/covered
```

File: code/modules/clothing/masks/breath.dm [hunk 4]
```diff
@@ -44,6 +44,7 @@
 	icon_state = "voxmask"
 	item_state = "voxmask"
 	permeability_coefficient = 0.01
+	flags_cover = null
```
Comment (Sirryan2002) [inline code/modules/clothing/masks/breath.dm:47]: this has other implications such as droppers and improvised garrotes.

**Review by Sirryan2002 — COMMENTED** [review]
I think this PR is falling into the mistaken idea that Realism === RP/Immersion. Space Station 13 is not a survival/milsim/realism based game (it originally being an atmos sim is not a valid argument), our game is based in fantasy, cartoon space violence, and magic. This feature will add tedium to an already boring feature and not provide a contribution to the roleplay enviroment as going the extra steps of taking off and putting a mask back on doesn't forward player character development or development the game world around the player.

**Comment by Coolrune206** [comment]
I dislike the concept of the PR _but_ it does kill Vox, so it can't be all bad.

How about we work away from the realism aspect and more into Vox dying, eh?

**Comment by hal9000PR** [comment]
> plasmamen need to be handled somehow for this to be a viable idea.

Plasmamen don't need to eat.

**Comment by Bm0n** [comment]
> > plasmamen need to be handled somehow for this to be a viable idea.
> 
> Plasmamen don't need to eat.

this is still a nerf to them eating pills or drinking chems from a beaker

**Comment by PacifistDalek** [comment]
Not a fan, this feels like it indirectly punishes space travel pretty harshly. And I definitely think that food is one area where comedy/shenanigans trumps realism when it comes to ss13.

**Comment by AffectedArc07** [comment]
Your PR has been put on hold as your GitHub account is not linked to a Paradise forums account.

Please link your account at [https://www.paradisestation.org/forum/custom/link_github_account.php](https://www.paradisestation.org/forum/custom/link_github_account.php)

*If you are seeing this multiple times due to multiple PRs you only have to link once*

**Comment by roarbark** [comment]
i like the underlying idea and ramifications of this. having to take off your mask and set it aside when you go to have lunch or something. i really vibe with that.

furthermore, concerning things like pills, you add more chance for skilled menu-ing and chances to fumble, which is what makes ss13's framework of interacting with the world special in a way- you dont just press e on a cigarette carton to smoke it, you have to do a non-linear set of actions to get it out of the box, lit, into your mouth-

and last point, if i was a new player who packed food for a space exploration and saw myself starving, decided to eat to realize i wont be able to through a hardsuit, that would be such a cool "wait what did i expect" moment

**Comment by Edan52** [comment]
This does not get me immersed in RP at all, it's just an annoying inconvenience.

Objecting.

**Comment by AffectedArc07** [comment]
No response from PR author in 7 days. Closing.

**Pull request closed without merging**